src/main.rs

```diff
@@ -1,42 +1,80 @@
 #![feature(proc_macro_hygiene, decl_macro)]
 
-mod payload;
-
 #[macro_use]
 extern crate rocket;
 
+use std::env;
+
+use rocket::config::{Config, Environment};
 use rocket::Data;
-use rocket::http::{Status, ContentType};
+use rocket::http::Status;
 use rocket::response::Content;
-use image::RgbaImage;
-use crate::payload::IntoVec;
+
+mod payload;
+mod blur;
 
 macro_rules! deny_out_of_range {
     ($expression:ident, $range:expr) => {
         if !$range.contains(&$expression) {
-            return Err(Status::Forbidden)
+            return Err(Status::BadRequest)
         }
     };
 }
 
-#[post("/blur?<sigma>", data = "<image>")]
-fn blur(image: Data, sigma: f32) -> Result<Content<Vec<u8>>, Status> {
-    let (image, _) = payload::image_from_data(image)?;
-    deny_out_of_range!(sigma, 0.0 .. 10.0);
+macro_rules! deny_smaller_than {
+    ($expression:ident, $smallest:expr) => {
+        if $smallest > $expression {
+            return Err(Status::BadRequest)
+        }
+    };
+}
 
-    let image: RgbaImage = imageproc::filter::gaussian_blur_f32(&image.into_rgba8(), sigma);
-    Ok(Content(ContentType::PNG, image.as_vec()?))
+macro_rules! deny_bigger_than {
+    ($expression:ident, $biggest:expr) => {
+        if $biggest < $expression {
+            return Err(Status::BadRequest)
+        }
+    };
 }
 
-#[post("/unsharpen?<sigma>&<threshold>", data = "<image>")]
-fn unsharpen(image: Data, sigma: f32, threshold: i32) -> Result<Content<Vec<u8>>, Status> {
-    let (image, _) = payload::image_from_data(image)?;
+#[post("/blur?<sigma>&<repeat>&<delay>", data = "<data>")]
+fn blur(data: Data, sigma: f32, repeat: i16, delay: u16) -> Result<Content<Vec<u8>>, Status> {
+    let image = payload::safe_vector_from_data(data)?;
+    let reader = payload::image_reader_from_data(&image)?;
+    let format = payload::image_format_from_reader(&reader)?;
     deny_out_of_range!(sigma, 0.0 .. 10.0);
+    deny_smaller_than!(repeat, -1);
+    deny_bigger_than!(delay, 2000);
 
-    let image = image.unsharpen(sigma, threshold);
-    Ok(Content(ContentType::PNG, image.as_vec()?))
+    let content = blur::blur_base_on_type(&image, reader, format, sigma, repeat, delay);
+    Ok(content)
 }
 
+
+// First argument supplied to the executable will be parsed and used as environment
```

Comment:
It doesn't matter if there's not, but isn't there any way to query what environment rocket _would_ pick at runtime? Try diving into the default configuration to see.

The reason is because this kind of environment choosing should really be done using a `Cargo.toml` file at runtime, and not an argument.